core/domain/subtopic_page_domain.py

```diff
@@ -30,73 +29,39 @@
+    """Domain object for changes made to subtopic_page object.
 
+    The allowed commands, together with the attributes:
+        - 'create_new' (with topic_id, subtopic_id)
+        - 'update_subtopic_page_property' (
+            with property_name, new_value, old_value, subtopic_id).
+    """
+
+    # The allowed list of subtopic page properties which can be used in
+    # update_subtopic_page_property command.
     SUBTOPIC_PAGE_PROPERTIES = (
         SUBTOPIC_PAGE_PROPERTY_PAGE_CONTENTS_HTML,
         SUBTOPIC_PAGE_PROPERTY_PAGE_CONTENTS_AUDIO,
         SUBTOPIC_PAGE_PROPERTY_PAGE_WRITTEN_TRANSLATIONS)
 
-    OPTIONAL_CMD_ATTRIBUTE_NAMES = [
-        'property_name', 'new_value', 'old_value', 'name', 'subtopic_id',
-        'topic_id'
-    ]
-
-    def __init__(self, change_dict):
-        """Initialize a SubtopicPageChange object from a dict.
-
-        Args:
-            change_dict: dict. Represents a command. It should have a 'cmd'
-                key, and one or more other keys. The keys depend on what the
-                value for 'cmd' is. The possible values for 'cmd' are listed
-                below, together with the other keys in the dict:
-                - 'update_topic_property' (with property_name, new_value
-                and old_value)
-
-        Raises:
-            Exception: The given change dict is not valid.
-        """
-        if 'cmd' not in change_dict:
-            raise Exception('Invalid change_dict: %s' % change_dict)
-        self.cmd = change_dict['cmd']
-
-        if self.cmd == CMD_UPDATE_SUBTOPIC_PAGE_PROPERTY:
-            if (change_dict['property_name'] not in
-                    self.SUBTOPIC_PAGE_PROPERTIES):
-                raise Exception('Invalid change_dict: %s' % change_dict)
-            self.property_name = change_dict['property_name']
-            self.new_value = copy.deepcopy(change_dict['new_value'])
-            self.old_value = copy.deepcopy(change_dict['old_value'])
-            self.id = change_dict['subtopic_id']
-        elif self.cmd == CMD_CREATE_NEW:
-            self.topic_id = change_dict['topic_id']
-        else:
-            raise Exception('Invalid change_dict: %s' % change_dict)
-
-    def to_dict(self):
-        """Returns a dict representing the SubtopicPageChange domain object.
-
-        Returns:
-            A dict, mapping all fields of SubtopicPageChange instance.
-        """
-        subtopic_page_change_dict = {}
-        subtopic_page_change_dict['cmd'] = self.cmd
-        for attribute_name in self.OPTIONAL_CMD_ATTRIBUTE_NAMES:
-            if hasattr(self, attribute_name):
-                subtopic_page_change_dict[attribute_name] = getattr(
-                    self, attribute_name)
-
-        return subtopic_page_change_dict
+    ALLOWED_COMMANDS = [{
+        'name': CMD_CREATE_NEW,
+        'required_attribute_names': ['topic_id', 'subtopic_id'],
+        'optional_attribute_names': ['change_affects_subtopic_page']
```

Comment:
Yes, I think we should do that since the attribute is required to find if the command belongs to topic_domain or subtopic_page_domain. Shall I go ahead with this change?

Reply:
I think so -- please confirm with @aks681 though. (I think you can ping him.)

Reply:
Isn't `change_affects_subtopic_page` more of a TopicChange thing, so that only those changes to a topic that require changes here are done here?
I don't think it was ever used in subtopic_page_domain....

Reply:
Yes, that is right @aks681. I checked this as well and it is only required to check whether the change object should be for `SubtopicPageChange` or `TopicChange`. 

As per our discussion on hangouts - we could either remove this completely or keep it as an optional attribute which is never used except in the controller for splitting. 

So, I felt it was better to remove this and directly find if the change command affects the subtopic page property i.e. 
```
change.cmd == subtopic_page_domain.CMD_UPDATE_SUBTOPIC_PAGE_PROPERTY
```

I have made the changes for the same in PR, PTAL. Thanks!